overlord/snapstate/progress.go

```diff
@@ -48,10 +50,21 @@ func NewTaskProgressAdapterLocked(t *state.Task) progress.Meter {
 func (t *taskProgressAdapter) Start(label string, total float64) {
 	t.label = label
 	t.total = total
+	t.Set(0.0)
 }
 
+// minimalProgress before we lock the state and set the task progress
+const minProgress = 0.2
+
 // Set sets the current progress
 func (t *taskProgressAdapter) Set(current float64) {
+	t.current = current
+	// check if we made at least "minProgress" before we lock the state
+	if current != 0.0 && (t.current/t.total)*100.0-(t.lastReported/t.total)*100.0 < minProgress {
```

Comment:
I find this a bit more readable:
```
const minProgress = 0.2 / 100.0
// check if we made at least "minProgress" before we lock the state
if current != 0.0 && (t.current - t.lastReported) / t.total < minProgress {
    return
}
```

Reply:
No strong opinion either way, but yours is shorter so it wins :)